src/components/common/Form.jsx

```diff
@@ -0,0 +1,77 @@
+import React, { Component } from 'react';
+import Autocomplete from 'react-autocomplete';
+
+class Form extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedAddress: '',
+      isCollectionScheduleTextHidden: true,
+      isAutoTextHidden: false,
+      dateFormat: 'D/M/YYYY',
+      scheduleType: {
+        trash: 'trash',
+        recycle: 'recycle',
+        leaf: 'leaf',
+      },
+      dayOfWeek: {
+        sunday: 0,
+        monday: 1,
+        tuesday: 2,
+        wednesday: 3,
+        thrusday: 4,
+        friday: 5,
+        saturday: 6,
+      },
+    };
+
+    this.handleClick = this.handleClick.bind(this);
+  }
+  handleClick(event) {
+    this.setState({
```

Comment:
this codes confusing me a little bit, what click is this method handling?

Reply:
rename to   this.handleClick = this.handleTryAnotherSearchClick.bind(this);

Reply:
so the pattern you are using here, the method names should match...thinking

```js
this.resetFormClick = this.bind.resetFormClick.bind(this);
```

would probably be the best name for this. The reason being is, we could change the text to be whatever we would want and this would still describe what is happening.